src/Microsoft.AspNet.Mvc.Razor/Compilation/ICompilationService.cs

```diff
@@ -13,11 +11,11 @@ public interface ICompilationService
         /// <summary>
         /// Compiles content and returns the result of compilation.
         /// </summary>
-        /// <param name="fileInfo">The <see cref="IFileInfo"/> for the Razor file that was compiled.</param>
+        /// <param name="fileInfo">The <see cref="RelativeFileInfo"/> for the Razor file that was compiled.</param>
         /// <param name="compilationContent">The generated C# content to be compiled.</param>
         /// <returns>
         /// A <see cref="CompilationResult"/> representing the result of compilation.
         /// </returns>
-        CompilationResult Compile(IFileInfo fileInfo, string compilationContent);
+        CompilationResult Compile(RelativeFileInfo fileInfo, string compilationContent);
```

Comment:
put better, how is move from `IFileInfo` to `RelativeFileInfo` related to avoiding exceptions in `CompilationResult`?